### DISCUSSION TITLE
Development stopped?

### DISCUSSION THREAD

**Discussion by v3ss0n** [dendronhq/dendron#3890] [Q&A] [answered]
Dendron, becoming my favorite tool for note taking but development seems stopped this year. Is this getting abandoned?

**Comment by aleksey-rowan** [comment]
Please, see [this announcement](https://discord.com/channels/717965437182410783/737323300967022732/1072563304289030164) on Dendron's Discord server.

**Reply by Binly42** [reply]
for guys without a Discord account, might refer to [this reddit post](https://www.reddit.com/r/dendron/comments/10z19pz/so_on_tuesday_the_developer_of_dendron_announced/) , and seems a sad story ...

**Comment by jangroth** [comment, marked as answer]
To save everyone following the links. Yes, Dendron development has stopped

This is @kevinslin announcemnt from February:

> @everyone
> 
> You might have noticed the slow down of activity from the team in discord and github over the last two months.
> I'm writing to give everyone an update on the status of Dendron. 
> 
> First - apologies for the radio silence. I've been dealing with some burnout as well as personal matters over the past few months that has made it hard for me to post on discord. While its not an excuse, it is just something I wanted to put out there for explanation.
> 
> In terms of Dendron, the tldr: we spent the past two years building a better way for humans to manage knowledge. While we made great strides there, as a business, we were ultimately not able to find product market fit for a venture backed business. After much soul searching, we've made the decision to do a pivot and pursue other business problems.
> 
> This means that Dendron, the extension, will no longer be actively developed. The good news is that Dendron is both local first and open source - this means that Dendron will continue to function. In fact, both myself and the team will continue to use Dendron (the entire reason I started Dendron was because I couldn't find a tool to manage all my notes - a fact that still remains true today). 
> 
> Dendron the project will be in maintenance mode moving forward. This means we won't be actively developing features (though we'll make a best effort basis on supporting existing features and reviewing pull requests). All recurring donations to Dendron have also been paused at this point (as we are no longer actively working on the project - thank you all for your contributions). 
> 
> As part of this announcement, we are also updating the Dendron license from "GPLV3" to "Apache 2.0". This is an even more permissive open source license and means that it will be even easier to fork, develop, and incorporate Dendron into your workflow (this was a frequent ask from our customers working at big tech companies).
> 
> Finally, I would like to dedicate the last part of this announcement to the Dendron community! Thank you all for being part of our adventure. 
> I often get comments from people that the community here is their favorite aspect about Dendron - some people that don't use Dendron still stay because of all the wonderful people here. 
> From answering questions to sharing workflows, thank you for making this such a great space! Special shout out to the following folks (and many many more)
> @foureyedsoul @benhsm | Jack of N trades @aleksey @ScriptAutomate @SeriousBug @Bassmann @Tika @tsnieman @cameron  @d1onys1us@rlh1994 
> 
> If you like to stay in touch, feel free to DM, follow on twitter (@kevins8), or message me at kevinlin@dendron.so. You can also stay in the loop via our mailing list here: https://buttondown.email/dendron

**Comment by v3ss0n** [comment]
If dendron was built , not as an vscode extension but as a dedicated app, it would have been a success.

**Reply by jangroth** [reply]
Not that it matters now, but I only picked it up because it _was_ a vscode extension. This slotted in very nicely into my workflow as well as allowing me to re-use all the vscode shortcuts I already know. Also easier to use on managed computers.

Currently switching to Obsidian and having a hard time relearning the keymap.

**Reply by Bassmann** [reply]
What makes you think so?

**Reply by v3ss0n** [reply]
Here are my thoughts:

1. Knowledge management tools are used by all sorts of fields , not just development , many research fields need it as much as we do , many office workers need it as much as we do. That would increase a lot of adotption. Now limited to a Single code editor and cannot port to anything else because Written in such a way that totally dependent on Vscode.
Wrapping it in VSCodium and use it as default application , releasing as a standalone binary would increase adoption by a lot.
I  tried to introduce this to several non-developer tech guys (tech founders , project managers) they won't even try coz it is a vscode extension.

2. Developers should had put effort on `need to have features` instead of `nice to have features` 
It is really hard to do basic features like PDF Direct Export , 
Instead they worked on Next base static web site (which is mostly broken).

**Reply by v3ss0n** [reply]
Try LogSeq  , it have great developement progress and growth even tho this was a lot cooler.

**Reply by dionysuzx** [reply]
the only gripe i have with dendron is two things:
1. hard to onboard non-engineers to use the tool (i think we can fix this with education, everyone ought to learn vscode in today's age honestly)
2. update the base web site (nextjs) into something that looks a bit nicer when compared to the other frontends like nextra, for example

if we can fix these i think it will be really great. i haven't looked into logseg @v3ss0n but does it support hierarchies and refactoring like dendron? this is the most powerful feature in my opinion. along with unique hashes for each note on the static page.

**Reply by v3ss0n** [reply]
> the only gripe i have with dendron is two things:
> 
> 1. hard to onboard non-engineers to use the tool (i think we can fix this with education, everyone ought to learn vscode in today's age honestly)
> 2. update the base web site (nextjs) into something that looks a bit nicer when compared to the other frontends like nextra, for example
> 
> if we can fix these i think it will be really great. i haven't looked into logseg @v3ss0n but does it support hierarchies and refactoring like dendron? this is the most powerful feature in my opinion. along with unique hashes for each note on the static page.

problem is , Code editor scares non-code people  (even though excel is a lot harder to use lol)
yeah i agree about 2. but should had based upon something with viet , that nextjs is so slow.

**Reply by dionysuzx** [reply]
i am willing to contribute my time to help build this. i'd like to use sveltekit and picocss

**Reply by v3ss0n** [reply]
> i am willing to contribute my time to help build this. i'd like to use sveltekit and picocss

do you mean rewriting from scratch?

**Reply by rixrix** [reply]
I've tried hard to use Dendron for months (Dec 2022), but I find it tricky to use -- I like the idea but yeah something's about the UX, not enough community plugins.

I've moved to Obsidian a month ago (before even hearing this announcement) .... it's like Dendron but on steroids, haven't tried other alternative so my opinion is a bit biased towards Obsidian - just copy/pasted my stuff. works!

**Comment by jangroth** [comment]
It would have been better for Dendron as an idea if Kevin had transitioned the project into a community project, instead o grinding it to a halt. 

Having said this, it probably wasn't an option given he's struggling with burnout. Truly and honestly wishing him all the best, I can only assume it's more than a github project on the line here.

**Reply by dionysuzx** [reply]
yeah agreed, there are probably a bunch of details, and surely a difficult time. an unforgettable project for me, built a great community around it. also wishing the best for the team.

the good news is that the project has an apache 2.0 license! so i think it's possible we can continue fixing the things that fix the community needs... for me that's really just a better frontend, and that's basically it. just need to take some time to learn the codebase...

**Reply by alexandrupetrini** [reply]
On my side, as DevOps Enginneer, multi feature freak and allways searching to find best or better all in one VSCode solution, I still find the interface a bit heavy loading when working with multiple project workspaces... 

But, in my opionion, the flop is in complexity. I don't see a lot of Geeks in 2 of the companies I've been so far, that would add it to their list of need to know.  Once, one of my teammates said he tried, for some time, to pitch others for Vivaldi browser (which I proudly use) and said, basically, that nobody wanted to hear about it. It's a solution for geeks or people that realy need it (also love this stuff). The others, go minimal, Keep It Simple Stupid, whatever it works, go on with life.

I switched to foam, see how I get on with it and it feels faster, for now.

Dendron was and it still looks a promicing idea, but it needs tons of fixes!

I had a lot of issues with:
- frontmatter import from Joplin, 
- tons of errors and frontmatter getting scrambled
- plus refactoring errors (literally stopping at 1'st-3'rd file to be renamed)

It's really hard to make it work in VSCode....

Sorry to be dissaponting, geeks love it, people who need it love it, I love it! 
It's hard to get it the right way though.....

**Reply by jangroth** [reply]
I moved to Obsidian. It's been a learning curve, obviously.

So far I've been blown away by the amount of available plugins, e.g. linting markdown and adding tags on save. Canvas is another nice feature, almost like a Miro board for my notes. Also many of my peers are already using Obsidian, which is nice. 

I do miss the VS-Code integration Dendron provided, but I can see that this is not a use case for everyone.

**Reply by alexandrupetrini** [reply]
Due to company policies, I usually need a free-4-all solution and cannot even buy or use an Obsidian licence without appropriate aproval.

**Reply by v3ss0n** [reply]
we could start if there are enough interest to maintain this repo. I propose 
 - Export to PDF Feature
 - Replace Next with Vite base builder , and a proper CSS layout
 - VSCodium based Desktop Application , where this is pacakged as default .

**Reply by jangroth** [reply]
> Due to company policies, I usually need a free-4-all solution and cannot even buy or use an Obsidian licence without appropriate aproval.

Obsidian is free ~and open source~. While you could pay for sync and publish features, this can easily be set up with community plugins and git integration.

**Reply by Bassmann** [reply]
That's not correct see https://obsidian.md/eula. If you use it for work related stuff you need a commercial license as explained also here https://help.obsidian.md/Licenses+and+payment/Commercial+license

**Reply by Bassmann** [reply]
> we could start if there are enough interest to maintain this repo. I propose
> 
>     * Export to PDF Feature
> 
Doesn't the export work for you that the Pandoc render extension creates?
>     * Replace Next with Vite base builder , and a proper CSS layout
> 
I don't use publishing but what's the issue with the existing option? 
>     * VSCodium based Desktop Application , where this is pacakged as default .
I'm not sure this eases the use of Dendron. One of the advantages of it for me is the vast set of extensions I can use and many of them are not available for VSCodium

**Reply by v3ss0n** [reply]
> > we could start if there are enough interest to maintain this repo. I propose
> > ```
> > * Export to PDF Feature
> > ```
> 
> Doesn't the export work for you that the Pandoc render extension creates?
> 
> > ```
> > * Replace Next with Vite base builder , and a proper CSS layout
> > ```
> 
> I don't use publishing but what's the issue with the existing option?
> 
> > ```
> > * VSCodium based Desktop Application , where this is pacakged as default .
> > ```
> 
> I'm not sure this eases the use of Dendron. One of the advantages of it for me is the vast set of extensions I can use and many of them are not available for VSCodium

1 - give a try, it takes a lot of steps not direct
2 - mostly broken
3 - all extensions are available just you need to add Ms registry

**Reply by Binly42** [reply]
> Obsidian is free and open source.

BTW [Obsidian is not open source](https://forum.obsidian.md/t/is-it-true-that-obsidian-is-already-open-source/46413/9)

**Reply by kbudde** [reply]
> > Obsidian is free and open source.
> 
> BTW [Obsidian is not open source](https://forum.obsidian.md/t/is-it-true-that-obsidian-is-already-open-source/46413/9)

What's hosted here in GitHub? I did not try to package it but it really looks like the source code of the dendron plugin. It's this not the case or is something missing?

**Reply by v3ss0n** [reply]
> > > Obsidian is free and open source.
> > 
> > 
> > BTW [Obsidian is not open source](https://forum.obsidian.md/t/is-it-true-that-obsidian-is-already-open-source/46413/9)
> 
> What's hosted here in GitHub? I did not try to package it but it really looks like the source code of the dendron plugin. It's this not the case or is something missing?

> Is obsidian code released with an open source license (MIT, GPL, etc) ? No.
> Is a (minified/obfuscated/packed) version of the code available within electron? Yes.

need to be resverse engineered , and not going to.

**Reply by kbudde** [reply]
Sorry, my answer somehow landed in the wrong thread 😁 obsidian is not open source.

**Comment by kbudde** [comment]
IMO the first thing to do is to find some people who are able and willing to continue development on the product.
Someone responsible for CI and QA would help as well (many github actions are failing).

So far I can see no real active fork. I have a bad feeling about the future right now 😢

**Reply by jangroth** [reply]
On top of that this is all a no-go without @kevinslin being part of the conversation.

**Reply by dphapsu** [reply]
Why? Dendron is not dead because Kevin has moved on to other things. It is dead only if nobody else picks it up. 

He developed this fantastic system for his own use and was kind enough to make it public but you can't expect a guy to work indefinitely on something that doesn't bring significant revenue. Also, since he uses it daily I assume that he will make updates as needed to keep it working for his use case so it should continue to work as is for the rest of us. If you want a new feature then fork it and dig into the code and write it. If you don't have the skill set then, GASP!, pay someone to code it for you. Required effort and money are the universes way of helping you decide if you really want something or if you just think you do.

In summary. Kevin, this is a wonderful system you have made. I know you have invested countless hours into it and I just want to say Thank You and wish you the best of luck on your other ventures.

**Reply by v3ss0n** [reply]
its better to have community is one place . If we fork and work on it , community split witll occur. It would be best if @kevinslin  can give maintainership to group of people by  : converting this into organizaiton and have multiple maintainers.

**Reply by dphapsu** [reply]
I absolutely agree. That is the best outcome. However,

1. If he does, GREAT! But he is not obligated to do so. 
2. Worst case scenario is he is no longer involved in Dendron. Even then it is possible to continue without him.  All other scenarios are improvements on this.

Question: Have people come forward who have the skill set to continue the project and are willing to devote the time and effort? I certainly hope so.

**Reply by jangroth** [reply]
Of course I agree that @kevinslin isn't obligated to do anything. 

However, I think at this stage the situation is confusing to say the least as the project is ghosted by its creator and sole maintainer. This extends to the website and the vs-code download. Obviously this has a tremendously negative impact on the community, as Dendron loses momentum with every day/week/month going by.

> Question: Have people come forward who have the skill set to continue the project and are willing to devote the time and effort? I certainly hope so.

How would you come forward when there's no communication? Forking the repo will only split the community further as already pointed out. Also, keep in mind Dendron is a lot more than a github repo - not having the ability to at least point the community to a new repo/updated website/changed VSCode plugin sounds like the perfect setup for failure/irrelevance to me.

The situation might have evolved differently had there been public conversation about transitioning the project into a community project, but that hasn't happen.

TL;DR: Without @kevinslin's involvement it's not realistic to transfer the project into something community driven. 

BTW, this is happening against the backdrop of @kevinslin being quite active on Twitter. Just observing.

**Reply by djradon** [reply]
Excerpted from my Discord comment...

There is no other piece of software that I value more! Other than Firefox, there's no software I use more! I am  so grateful to @kevins8 and his team and the once-great community-at-large for giving it a go. I can't pick up development myself right now, but if anyone ever needs a project manager for a Dendron reincarnation, please let me know.

**Comment by neosartre-nyc** [comment]
All,

I just found Dendron. I believe the opportunity is there to continue what was started. I will put my thoughts together. If anyone has serious ideas about this, please let me know.

NeoSartre

**Comment by balupton** [comment]
Just found Dendron, I'm guessing Foam is the alternative: https://github.com/foambubble/foam

**Reply by neosartre-nyc** [reply]
I haven’t taken a look yet, but I am interested. I am currently using Zettlr. It’s good for publishing, but I struggle with PKMS’s in general these days. I just want something that’s simple as in open source, files accessible, now soul sucking eco-systems.

**Reply by v3ss0n** [reply]
dendron was alternative of foam. Dendron was better in the way it organize things.
